In [2]:
import pandas as pd

df = pd.read_csv("content_refresh_anonymized.csv") 

print("Rows:", len(df))
print("Columns:", len(df.columns))

print("\nAverage CTR:", df["ctr"].mean())
print("Average Position:", df["avg_position"].mean())

print("\nContent Types:")
print(df["content_type"].value_counts())

print("\nTrend Direction:")
print(df["trend_direction"].value_counts())

Rows: 30000
Columns: 44

Average CTR: 0.5107333333333334
Average Position: 16.342380000000002

Content Types:
content_type
keyword article       27207
feedly article         2096
comparison article      697
Name: count, dtype: int64

Trend Direction:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


# My Lane

I selected the **Content Optimization** predefined lane because the dataset contains webpage performance metrics such as impressions, clicks, CTR, average search position, content age, and trend direction. These features can help identify which webpages should be prioritized for optimization. The goal is to support better SEO and content decisions rather than simply building a machine learning model.

# The Question

## Research Question

Which webpages should be prioritized for optimization to improve their organic search performance?

## Decision

Determine which webpages deserve attention first.

## Unit of Analysis

One webpage.

## Output

A prediction or ranking showing whether a webpage should be prioritized for optimization.

## Action

The SEO or content team can update outdated pages, improve titles and descriptions, refresh content, or improve internal linking for the recommended pages.

## Cost of a Wrong Recommendation

A false positive wastes time and resources on pages that may not improve.

A false negative ignores pages that could generate more traffic if optimized.

## Why Machine Learning?

Many variables influence webpage performance simultaneously. Machine learning can identify patterns across these variables and help prioritize pages more effectively than manual inspection.

# Quick Look at the Data

The starter dataset contains **30,000 webpages** and **44 features**.

The average click-through rate (CTR) is **0.5107**, while the average search position is **16.34**.

Most pages are **keyword articles (27,207)**, followed by **Feedly articles (2,096)** and **comparison articles (697)**.

The trend distribution shows that **16,262 pages are trending down**, compared to **4,388 trending up**, suggesting many pages may benefit from content optimization. This supports the choice of the Content Optimization lane.

# Careful Words

This project does not prove that updating a webpage will increase traffic.

Instead, it identifies pages that appear to have a higher likelihood of benefiting from optimization based on historical data.

The recommendations should be treated as decision support rather than guarantees because webpage performance is also influenced by search intent, competition, seasonality, and search engine algorithm changes.

# Self Check

- Selected a predefined lane
- Defined the research question
- Explained the decision and action
- Described the cost of wrong recommendations
- Included real numbers from the dataset
- Explained why machine learning is useful
- Avoided making causal claims